addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/automation/ui/ScriptJobDialog.java

```diff
@@ -56,42 +73,119 @@ public ScriptJobDialog(ScriptJob job) {
+            if (disabledFields.contains(fieldName)) {
+                this.getField(fieldName).setEnabled(false);
+                this.setFieldValue(fieldName, "");
+            } else {
+                this.getField(fieldName).setEnabled(true);
+            }
+        }
     }
 
     private void onScriptTypeChanged() {
         String scriptType = this.getStringValue(SCRIPT_TYPE_PARAM);
         List<String> scripts = ScriptAction.getAvailableScriptNames(scriptType);
+        // Always have a blank option at the start
+        scripts.add(0, "");
         this.setComboFields(
                 SCRIPT_NAME_PARAM, scripts, this.job.getData().getParameters().getName());
     }
 
+    private void onScriptNameChanged() {
+        ScriptAction sa = getScriptAction();
+        if (sa.getDisabledFields().contains(SCRIPT_FILE_PARAM)) {
+            String scriptName = this.getStringValue(SCRIPT_NAME_PARAM);
+            ScriptWrapper sw = ScriptAction.getExtScript().getScript(scriptName);
+            if (sw != null) {
+                this.setFieldValue(SCRIPT_FILE_PARAM, sw.getFile().getAbsolutePath());
+            }
+        }
+    }
+
     @Override
     public void save() {
         this.job.getData().setName(this.getStringValue(NAME_PARAM));
         this.job.getData().getParameters().setAction(this.getStringValue(SCRIPT_ACTION_PARAM));
         this.job.getData().getParameters().setType(this.getStringValue(SCRIPT_TYPE_PARAM));
+        this.job.getData().getParameters().setEngine(this.getStringValue(SCRIPT_ENGINE_PARAM));
         this.job.getData().getParameters().setName(this.getStringValue(SCRIPT_NAME_PARAM));
+
+        ScriptAction sa = getScriptAction();
+        if (sa.getDisabledFields().contains(SCRIPT_FILE_PARAM)) {
+            this.job.getData().getParameters().setFile(null);
+        } else {
+            File f = new File(this.getStringValue(SCRIPT_FILE_PARAM));
+            if (f.isFile()) {
+                this.job.getData().getParameters().setFile(this.getStringValue(SCRIPT_FILE_PARAM));
+            }
+        }
         this.job.resetAndSetChanged();
     }
 
     @Override
     public String validateFields() {
-        return null;
+        ScriptAction sa = getScriptAction();
+        if (sa.getDisabledFields().contains(SCRIPT_FILE_PARAM)) {
+            // Always unset this as the text field cannot be edited
+            this.setFieldValue(SCRIPT_FILE_PARAM, null);
+        }
+        ScriptJobParameters params =
+                new ScriptJobParameters(
+                        this.getStringValue(SCRIPT_ACTION_PARAM),
+                        this.getStringValue(SCRIPT_TYPE_PARAM),
+                        this.getStringValue(SCRIPT_ENGINE_PARAM),
+                        this.getStringValue(SCRIPT_NAME_PARAM),
+                        this.getStringValue(SCRIPT_FILE_PARAM));
+        sa = ScriptJob.createScriptAction(params, null);
+        List<String> issues = sa.verifyParameters(this.job.getName(), params, null);
```

Comment:
Should use the current job name, otherwise the warning is confusing when the name is changed.